### PR TITLE
Improving Javascript style

### PR DESCRIPTION
**Pull request by Vandan-V-Phadke**
As per our conversation I am sending this PR.
Please guide me if I am not going in the right direction.

### CODE REVIEW

File: Source/background.js [hunk 1]
```diff
@@ -1,38 +1,45 @@
...
+        storage.set(data);
+        }
+    });
 }
 initializeStorage();
 
-
+/**
+ * @desc Constructs google translate url
+ * @params User saved preferences
```
Comment (Ceilican) [inline Source/background.js]: here you mean "prefs", right?

Comment (Ceilican) [inline Source/background.js]: From google's style guide:

```
/**
 * Some class, initialized with a value.
 * @param {Object} value Some value.
 * @constructor
 */
function MyClass(value) {
  /**
   * Some value.
   * @type {Object}
   * @private
   */
  this.myValue_ = value;
}
```

**Comment by Ceilican** [comment]
Thanks for the pull request, Vandan! I have just merged it.

You are going in the right direction, but a few things are not according to Google's style guidelines for Javascript (https://google-styleguide.googlecode.com/svn/trunk/javascriptguide.xml). For example, 2 spaces should be used for code block indentation, while 4 spaces should be used for statement break indentation. Have a look on the style guide, and you will see many examples of good style there.

**Comment by Ceilican** [comment]
Also, follow git's convention and use imperative instead of gerund in your commit messages. For example, among git users, it would be generally considered better to write "Improve Javascript style" than "Improving Javascript style".

**Comment by Vandan-V-Phadke** [comment]
Thanks a lot ...
I will take these things into consideration and send the next pull request.

I had a some small doubts in mind

And what is the function of the ngrammin and ngramMax params in getAllWords
method?
Is POS tagging already integrated ?
How exactly are we selecting which words to translate and in which method
are we filtering it  ?

Vandan Phadke

**Comment by Ceilican** [comment]
By default, MindTheWord translates isolated words.  However, there is an advanced option in the options page that allows the user to choose ngram-min and ngram-max. When the user selects, for example, ngram-min = 2 and ngram-max = 4, MindTheWord will translate sequences of consecutive words containing from 2 to 4 words.

**Comment by Ceilican** [comment]
POS tagging is currently not used. If we had it, we could easily implement the following requested feature: https://github.com/OiWorld/MindTheWord/issues/30

The difficulty are:
1) POS tagging tools are probably not available for all languages.
2) relying on a web-service that provides POS tagging would probably be too slow.

If you have any idea how how to overcome these challenges in a simple way, I would really like to hear it!

**Comment by Ceilican** [comment]
The text is broken into a collection of words (or n-grams/wordsequences) in the method ```getAllWords```.  Perhaps "getAllWordSequences" would be a better name for this method.

Then the words are filtered in ```filterSourceWords```.  Issue 37 (https://github.com/OiWorld/MindTheWord/issues/37) seems to be affecting this method. For some reason, it translates much fewer words than it should, when given high values of translation probability.

**Comment by Ceilican** [comment]
Have a look also on the ```shuffle``` method. It is responsible for shuffling the list of words extracted from the text, so that we don't end up translating always the same words.

**Pull request merged**